pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -161,11 +147,17 @@ def process_dataloader(self, dataloader: DataLoader) -> MpDeviceLoader:
     def configure_ddp(self) -> None:
         pass
 
-    def init_dist_connection(self, global_rank: int, world_size: int) -> None:
-        pass
-
     def set_world_ranks(self, process_idx: int = 0) -> None:
-        pass
+        # self._local_rank = process_idx
+        #
+        # if self.cluster_environment is None:
+        #     return
+        # self.cluster_environment.set_global_rank(self.node_rank * self.num_processes + self.local_rank)
+        # self.cluster_environment.set_world_size(self.num_nodes * self.num_processes)
+        # self.cluster_environment.set_global_rank(xm.get_ordinal())
+        # rank_zero_only.rank = self.cluster_environment.global_rank()
+        super().set_world_ranks(xm.get_local_ordinal())
```

Comment:
@awaelchli Could I be of any help here?

Reply:
@kaushikb11 Yes that would be great! I'm wondering if `process_idx == xm.get_local_ordinal()` is true or if I'm missing something about these definitions here.

Reply:
Yup, it should be true